TheEthicalHackerCup/Assets/DefenseMinigames/Ransomware/dRansomManager.cs

```diff
@@ -0,0 +1,190 @@
+
+        updateHeuristics(heuristics);
+        updateAll();
+
+    }
+
+    private void updateHeuristics(string heuristics)
+    {
+        heuristicText.text = heuristics;
+    }
+
+    private void updateMetaData() 
+    {
+        fileText.text = current.ToString();
+    }
+
+    private void updateLives() {
+        this.livesText.text = "Lives: " + lives;
+    }
+
+    private void updateSandboxStatus() 
+    {
+        if(!this.timerOn) 
+        {
+            this.sandboxStatusText.text = "";
+            return;
+        }
+        sandboxStatusText.text = this.current.isVirus && this.current.TimeTillInfection <= timer?"Virus Detected":"Program Sandboxed";
+    }
+
+    private void updateTimer() {
+        this.timerText.text = timerOn ? "Time program is in sandbox: "+timer : "";
+    }
+
+    private void Update()
+    {
+        if (this.timerOn) 
+        {
+            timer += Time.deltaTime;
+            updateTimer();
+            updateSandboxStatus();
+        }
+    }
+
+    private void handleAcceptClicked() 
+    {
+        if (current.isVirus)
+        {
+            lives-=2;
+            updateLives();
+            checkForDeath();
+        }
+        nextData();
+    }
+    private void handleDenyClicked() 
+    {
+        if (!current.isVirus)
+        {
+            lives--;
+            updateLives();
+            checkForDeath();
+        }
+        nextData();
+    }
+    private void handleSandboxClicked() 
+    {
+        if (timerOn) 
+        {
+            updateTimer();
+        }
+        timerOn = !timerOn;
+        updateSandboxStatus();
+    }
+
+    private void updateAll() 
+    {
+        
+        updateMetaData();
+        updateLives();
+        updateTimer();
+        updateSandboxStatus();
+    }
+
+    private void checkForDeath() 
+    {
+        if (lives <= 0) 
+        {
+            GameManager.GetInstance().SwitchToAfterActionReportScene("All your lives were lost. You either let ransomers destroy the all the production data (including backups), thought too many things were viruses or a combination of both.");
+            return;
+        }
+    }
+
+    private void nextData() 
+    {
+        if (dataIndex+1 == metaData.Count) 
+        {
+            GameManager.GetInstance().SwitchToAfterActionReportScene("You successfully scanned the incoming software and kept the production data safe. Good job!");
+            FileHelperSingleton.Singleton.nextFile();
+            return;
+        }
+        dataIndex++;
+        timer = 0;
+        timerOn = false;
+        updateAll();
+    }
+
+}
+
+class FileHelperSingleton
+{
+    private string[] fileNames = { "RansomDefense/level1", "RansomDefense/level2", "RansomDefense/level3" };
+    private int index = 0;
+    public string currentFileName { get { return fileNames[index]; } }
+
+    public void nextFile() 
+    {
+        if (fileNames.Length == index + 1) 
+        {
+            return;
+        }
+        index++;
```

Comment:
After the player's 3rd time playing the game, would they always see the same level?

Reply:
Consider changing the index system here with a call to `GameManager.GetInstance().GetDefenseUpgradeLevel(SecurityConcepts.Ransomeware)` so that the level that gets loaded corresponds to the player's upgrade level

Reply:
i think imma keep it as is.
after the 3rd it stays the same